src/AuditTrailModule/javasource/audittrail/log/CreateLogObject.java

```diff
@@ -296,115 +292,141 @@ private static IMendixObject createSingleLogLine(IMendixObject logObject, IMendi
 						(Integer) logObject.getValue(context, Log.MemberNames.NumberOfChangedMembers.toString()) + 1);
 			}
 
-			return logLine;
+			return Collections.singletonList(logLine);
 		}
 
 		_logNode.trace("Skipping member: " + member.getName() + " because it has not changed.");
-		return null;
+		return Collections.emptyList();
 	}
 
-	private static IMendixObject createReferenceLogLine(IMendixObject logObject, IMendixObjectMember<?> member,
+	private static List<IMendixObject> createReferenceLogLine(IMendixObject logObject, IMendixObjectMember<?> member,
 			boolean isNew, IContext sudocontext, IContext currentcontext) throws CoreException {
 		// get current and previous id
 		IMendixIdentifier cID = (IMendixIdentifier) member.getValue(currentcontext);

```

Comment:
While we are at it lets call these variables `currentId` and `previousId`.

Reply:
Done